core/src/apps/cardano/sign_tx.py

```diff
@@ -316,7 +293,42 @@ def _validate_metadata(metadata: Optional[bytes]) -> None:
         raise INVALID_METADATA
 
 
-def _serialize_tx(keychain: seed.Keychain, msg: CardanoSignTx) -> Tuple[bytes, bytes]:
+def _get_signed_tx_chunks(
+    cborized_tx: CborizedSignedTx, tx_hash: TxHash
+) -> Iterator[CardanoSignedTx]:
+    remaining_len = 0
+    for token in cbor.encode_chunked(cborized_tx):
```

Comment:
this seems weird. Why do we need to know remaining_len at all? Can't we just iterate until the end?

Reply:
I found knowing the remaining length is convenient for allocation of a buffer of the exact size that is needed - relevant for the last chunk. But it's also valid to just reallocate a bytearray of the right size when StopIterator error is raised - if the chunk size is not too big (e.g. 1-2 kB), the additional overhead of allocating one more buffer is acceptable.

Another alternative would be returning the full chunk of fixed size and a length property alongside it but I find it weird to return redundant data and then instructing the client to trim them.